dmoj/tests/test_filesystem_policy.py

```diff
@@ -0,0 +1,53 @@
+import unittest
```

Comment:
Some more tests:

- too many /s, e.g. `/etc/////passwd` and `////etc/passwd`
- empty string

Reply:
Should the rules themselves call `os.realpath` to avoid this madness? Or do we want this as a sanity check

Reply:
I think we could move the call to `os.path.normpath` from `isolate.py` into `check`. Then isolate just needs to worry about obtaining _some_ valid path, and we can normalize it here where it's unit-testable. @quantum5 wdyt?

Reply:
Would we still need a check for an absolute path... probably, unless `check` takes two paths

Reply:
We could do `normpath`, yes, but what about conversion to absolute paths? That still needs to be handled by the caller. I guess we can have the caller do the join...

Also `os.path.realpath` resolves symlinks, and I am not sure if that's what we want. I think what we really want is to test both the original path and the path after resolving symlinks to avoid crazy attacks that way.

Reply:
Yes, let's just raise if a non-abspath is passed to `check`.

Reply:
On second thought, the only sane solution is to test both `normpath` and `realpath`. To deal with trickery, you can test to see if they are the same file with `samefile`.